src/Uno.Extensions.Storage.UI/FileStorage.cs

```diff
@@ -11,10 +13,45 @@ public class FileStorage : IStorage
 	private async Task<bool> FileExistsInPackage(string filename)
 	{
 #if __ANDROID__
-		var assets = global::Android.App.Application.Context.Assets;
-		var files = assets?.List("");
-		filename = Path.GetFileNameWithoutExtension(filename).Replace('.', '_') + Path.GetExtension(filename);
-		return files?.Contains(filename)??false;
+		//Look in assets first***
+		var context = global::Android.App.Application.Context;
+		var assets = context.Assets;
+		var normalizedFileName = Path.GetFileNameWithoutExtension(filename).Replace('.', '_').Replace('-','_') + Path.GetExtension(filename);
```

Comment:
Can we use the EncodeResourcePath from https://github.com/unoplatform/uno/blob/43ca40a9f9c86160d85fe346fe74682c2ae5da4a/src/Uno.UWP/Helpers/AndroidResourceNameEncoder.cs#L53
cc @jeromelaban would it be worth exposing this method so it can be used by extensions etc?

Reply:
That's a good point. I think we should be moving this code to uno, as an extension, as it relies on internal implementation details about the way resources are encoded.

Having something like `Uno.Storage.StorageFileHelper.Exists()` would be interesting instead.